pkg/addon/parameters.go

```diff
@@ -5,74 +5,92 @@ import (
 	"fmt"
 	"strconv"
 
+	l "github.com/integr8ly/integreatly-operator/pkg/resources/logger"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
 	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/apis/v1alpha1"
-	rhmiresources "github.com/integr8ly/integreatly-operator/pkg/resources/rhmi"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	QuotaParamName      = "addon-managed-api-service"
 	TrialQuotaParamName = "trial-quota"
+	ManagedAPIService   = "managed-api-service"
+	RHMI                = "rhmi"
+	DefaultSecretName   = "addon-managed-api-service-parameters"
+)
+
+var (
+	log = l.NewLoggerWithContext(l.Fields{l.ComponentLogContext: "addon"})
 )
 
-// GetParameter retrieves the value for an addon parameter by finding the RHMI
-// CR and selecting the addon name for its installation type.
+// GetParameter retrieves the value for an addon parameter by finding the Subscription
+// CR and selecting the addon name from a secret associated with it.
 func GetParameter(ctx context.Context, client k8sclient.Client, namespace, parameter string) ([]byte, bool, error) {
-	rhmi, err := rhmiresources.GetRhmiCr(client, ctx, namespace, log)
+	secret, err := GetAddonParametersSecret(ctx, client, namespace)
 	if err != nil {
 		return nil, false, err
 	}
 
-	return GetParameterByInstallation(
-		ctx,
-		client,
-		rhmi,
-		parameter,
-	)
+	value, ok := secret.Data[parameter]
+	return value, ok, nil
 }
 
-func GetParameterByInstallation(ctx context.Context, client k8sclient.Client, install *integreatlyv1alpha1.RHMI, parameter string) ([]byte, bool, error) {
-	return GetParameterByInstallType(
+// GetAddonParametersSecret retrieves addon parameters secret, provided operator namespace.
+// If namespace is not hive managed, looks for the secret name in envars
+func GetAddonParametersSecret(ctx context.Context, client k8sclient.Client, namespace string) (*corev1.Secret, error) {
+	parametersSecretName := DefaultSecretName
+
+	isHiveManaged, err := OperatorIsHiveManaged(
 		ctx,
 		client,
-		integreatlyv1alpha1.InstallationType(install.Spec.Type),
-		install.Namespace,
-		parameter,
-	)
-}
+		&integreatlyv1alpha1.RHMI{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+			},
+		})
+	if err != nil {
+		return nil, err
+	}
 
-// GetParameterByInstallType retrieves the value for an addon parameter by
-// selecting the addon name for installationType
-func GetParameterByInstallType(ctx context.Context, client k8sclient.Client, installationType integreatlyv1alpha1.InstallationType, namespace, parameter string) ([]byte, bool, error) {
-	addonName := GetName(installationType)
-	secretName := fmt.Sprintf("addon-%s-parameters", addonName)
+	if isHiveManaged {
```

Comment:
Following discussion. 
Now it would try to retrieve the subscription. If it finds one - use the name of the subscription as a name of a secret. If it fails to find the subscription it uses const `addon-managed-api-service-parameters`. 

Not, the fun with subs. We have 3 names of subscriptions: 
- `addon-managed-api-service` - addon. Secret name `addon-managed-api-service-parameters`
- `managed-api-service` - olm and sandbox. Secret name `addon-managed-api-service-parameters`
- `addon-managed-api-service-internal` - internal. Secret name `addon-managed-api-service-internal-parameters`
As you can see the code will now check for the `addon` at the beginning of the sub and if it doesn't have - the `addon` will be added. This approach also leaves room for the Sandbox people to update the function once they change the name for the MT subscription. 
This covers us for all installation types. I've also extended unit tests to properly simulate all 4 cases (3 subs and a local install)